### ISSUE TITLE
osquery creates multiple copies of the same extension

### ISSUE THREAD

**Issue by a-cognet** [osquery/osquery#7171]
# Bug report

### What operating system and version are you using?

```
osqueryi --line "SELECT version, build, platform FROM os_version;"
 version = CentOS Linux release 7.9.2009 (Core)
   build =
platform = rhel
```

### What version of osquery are you using?

```
osqueryi --line "SELECT version from osquery_info;"
version = 4.9.0
```

### What steps did you take to reproduce the issue?

The problem happens when the extension watchdog considers the extension not sane and restarts a new one without killing the original extension.

Setup:

Flags used explicitely:
```
--disable_extensions=false
--extensions_timeout=10
--extensions_interval=10
--extensions_autoload=/etc/osquery/extensions.load
--disable_watchdog=true
--watchdog_level=1
--verbose=true
--watchdog_utilization_limit=2
```
I increased `watchdog_level` and lowered `watchdog_utilization_limit` to ease the reproduction, but since their default value are non-zero, it does not change the behavior.

The flag `enable_extensions_watchdog` [defaults to false](https://github.com/osquery/osquery/blob/master/osquery/core/watcher.cpp#L108-L111)

Steps:
1. start osqueryd with an extension - get the pid of the extension
2. increase the extension cpu or memory usage so that watchdog limits are passed
3. observe the issue:
     - loglines `watcher.cpp:624] Extension respawning too quickly` and `watcher.cpp:658] Created and monitoring extension child` will appear in loops, showing multiple copies are created.
     - getting the pids again, the original extension is still alive (our extension is not able to run multiple copies successfully, se we do not see multiple extensions present)

### What did you expect to see?
Nothing: the watchdog is disabled
The documentation is explicit https://osquery.readthedocs.io/en/stable/installation/cli-flags/

### What did you see instead?
Osquery creating multiple copies of an extension passing the watchdog limits

Looking into it: the behavior of https://github.com/osquery/osquery/blob/master/osquery/core/watcher.cpp#L305 does not seem correct, we fixed it by using this patch https://github.com/osquery/osquery/compare/master...a-cognet:watchdog_restart_extensions

**Comment by directionless** [comment]
Was this fixed by #7178 ?

**Comment by a-cognet** [comment]
Yes, it was, thank you for checking.
